packages/verkle/src/verkleTree.ts

```diff
@@ -213,9 +215,149 @@ export class VerkleTree {
    * @param value - the value to store
    * @returns A Promise that resolves once value is stored.
    */
-  // TODO: Rewrite following logic in verkle.spec.ts "findPath validation" test
-  async put(_key: Uint8Array, _value: Uint8Array): Promise<void> {
-    throw new Error('not implemented')
+  async put(key: Uint8Array, value: Uint8Array): Promise<void> {
```

Comment:
This function is very long.  I find functions more difficult to read and debug when I have to scroll in order to see different parts of it.

I appreciate the way you use the comments to outline how things are happening step by step.  I wonder if these different steps could be externalized into separate functions, so that the `put` function itself can fit on a screen?

Reply:
This is a 100% valid criticism.  Lemme look at the logic and see if there's a way to break it down.  I think you're right that it can be done in smaller chunks.